components/SignInOrJoinFree.js

```diff
@@ -198,7 +198,14 @@ class SignInOrJoinFree extends React.Component {
             onSubmit={values => {
               const { twoFactorAuthenticatorCode, recoveryCode } = values;
               if (recoveryCode) {
-                return this.props.submitRecoveryCode(recoveryCode);
+                return this.props.submitRecoveryCode(recoveryCode).then(user => {
+                  if (user?.collective?.slug) {
```

Comment:
No need to change for now but in which case are we missing `user?.collective?.slug`? Why are we defensive here?

Reply:
@znarf  I added it in case the user provides an invalid recovery code, but I think we can remove it as `then` will run only if the `submitRecoveryCode` is resolved successfully.

Reply:
Updated 👍🏻